### DISCUSSION TITLE
authentication failure

### DISCUSSION THREAD

**Discussion by Qrangel1** [community/community#37034] [New to GitHub]
Hi, I have been trying to clone a Github repository to my linux terminal "Kali Linux". It asks me for my username and password, but then it tells me that the authentication failed, I don't know how to solve it. 

![IMG_20221023_083551](https://user-images.githubusercontent.com/116478283/197395347-9bb6542c-b773-4078-b0a2-960cb2103519.jpg)

**Comment by DSDmark** [comment]
Hey @Qrangel1, 
You need to create a **personal access token,** then it works ok. I will tell you all the steps of how you generate that token.
So, after you're done creating that simpler try again to clone the repo that you want and if it asks for **username** then simply give your username but, in the case of the password, you should use that **token** which you created ok.

So, here are the steps on how you created your token:-

---

> Step 1: Open your settings:-

- ![Screenshot from 2022-10-23 19-44-42](https://user-images.githubusercontent.com/85744816/197397500-31c74f36-7c02-40b7-abef-f78a0590f2a3.png)

---

> Step 2: Scroll down and click on developer settings

-![Screenshot from 2022-10-23 19-45-04](https://user-images.githubusercontent.com/85744816/197397559-854eac1a-0026-4d54-9173-0d2f625d081a.png)

---

> Step 3: After that, there is a **personal access token** section click on that and click on generate a new token.

- ![Screenshot from 2022-10-23 19-46-20](https://user-images.githubusercontent.com/85744816/197397648-d590d9fb-fd69-411f-8004-50a5b5bc25ae.png)

---

> Step 4:Click on expiration section and select no expiration.

![Screenshot from 2022-10-23 19-47-15](https://user-images.githubusercontent.com/85744816/197397633-fd39e3f6-2502-4f68-8927-8552282a7933.png)

---

> Step 5: Scroll down and click gernated new token.

- ![Screenshot from 2022-10-23 19-47-28](https://user-images.githubusercontent.com/85744816/197397823-0022f48a-caa6-44d8-b3a3-c4f62e9c9a0d.png)

---

> After all these steps, you can see your token. Use your token Instead of your password.

**Reply by DSDmark** [reply]
If you are curious why we create the token things, then I suggest you read this:-
[**Git password authentication is shutting down**](https://github.blog/changelog/2021-08-12-git-password-authentication-is-shutting-down/)

**Comment by ossanna16** [comment]
Hi there @Qrangel1 and welcome to our community 😄. Thanks so much for asking a great question!
